tribe-ext-pdf-tickets/index.php

```diff
@@ -0,0 +1,892 @@
+
+		if ( empty( $ticket_instance ) ) {
+			return $successful;
+		}
+
+		// should only be one result
+		$event_ids = tribe_tickets_get_event_ids( $attendee_id );
+
+		if ( ! empty( $event_ids ) ) {
+			$event_id  = $event_ids[0];
+			$attendees = $ticket_instance->get_attendees_array( $event_id );
+		}
+
+		if (
+			empty( $attendees )
+			|| ! is_array( $attendees )
+		) {
+			return $successful;
+		}
+
+		$attendees_array = array();
+
+		foreach ( $attendees as $attendee ) {
+			if ( $attendee['attendee_id'] == $attendee_id ) {
+				$attendees_array[] = $attendee;
+			}
+		}
+
+		if ( empty( $attendees_array ) ) {
+			return $successful;
+		}
+
+		/**
+		 * Because $html is the full HTML DOM sent to the PDF generator, adding
+		 * anything to the beginning or the end would likely cause problems.
+		 *
+		 * If you want to alter what gets sent to the PDF generator, follow the
+		 * Themer's Guide for tickets/email.php or use that template file's
+		 * existing hooks.
+		 *
+		 * @link https://theeventscalendar.com/knowledgebase/themers-guide/#tickets
+		 */
+		$html = $ticket_instance->generate_tickets_email_content( $attendees_array );
+
+		if ( empty( $html ) ) {
+			return $successful;
+		}
+
+		$file_name = $this->get_pdf_path( $attendee_id );
+
+		if ( empty( $file_name ) ) {
+			return $successful;
+		}
+
+		if ( file_exists( $file_name ) ) {
+			$successful = true;
+		} else {
+			$successful = $this->output_pdf( $html, $file_name );
+		}
+
+		/**
+		 * Action hook after the PDF ticket gets created.
+		 *
+		 * Might be useful if you want the PDF file added to the Media Library
+		 * via wp_insert_attachment(), for example.
+		 *
+		 * @var $ticket_class
+		 * @var $event_id
+		 * @var $attendee_id
+		 * @var $file_name
+		 */
+		do_action( 'tribe_ext_pdf_tickets_uploaded_pdf', $ticket_class, $event_id, $attendee_id, $file_name );
+
+		/**
+		 * Filter to disable PDF email attachments, either entirely (just pass
+		 * false) or per event, attendee, ticket type, or some other logic.
+		 *
+		 * @var $email
+		 * @var $ticket_class
+		 * @var $event_id
+		 * @var $attendee_id
+		 * @var $file_name
+		 */
+		$email = (bool) apply_filters( 'tribe_ext_pdf_tickets_attach_to_email', $email, $ticket_class, $event_id, $attendee_id, $file_name );
+
+		if (
+			true === $successful
+			&& true === $email
+		) {
+			$this->attachments_array[] = $file_name;
+
+			if ( 'Tribe__Tickets__RSVP' === $ticket_class ) {
+				add_filter( 'tribe_rsvp_email_attachments', array( $this, 'email_attach_pdf' ) );
+			} elseif ( 'Tribe__Tickets_Plus__Commerce__WooCommerce__Main' === $ticket_class ) {
+				add_filter( 'tribe_tickets_plus_woo_email_attachments', array( $this, 'email_attach_pdf', ) );
+			} elseif ( 'Tribe__Tickets_Plus__Commerce__EDD__Main' === $ticket_class ) {
+				add_filter( 'edd_ticket_receipt_attachments', array( $this, 'email_attach_pdf' ) );
+			} else {
+				// unknown ticket type so no emailing to do
```

Comment:
Optional: add a `do_action()` call so an enterprising developer could add support for the missing type?